api/src/main/java/org/pmiops/workbench/api/DiskController.java

```diff
@@ -0,0 +1,122 @@
+package org.pmiops.workbench.api;
+
+import java.util.List;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+import javax.inject.Provider;
+import org.pmiops.workbench.db.dao.WorkspaceDao;
+import org.pmiops.workbench.db.model.DbUser;
+import org.pmiops.workbench.db.model.DbWorkspace;
+import org.pmiops.workbench.exceptions.NotFoundException;
+import org.pmiops.workbench.leonardo.model.LeonardoDiskStatus;
+import org.pmiops.workbench.leonardo.model.LeonardoGetPersistentDiskResponse;
+import org.pmiops.workbench.leonardo.model.LeonardoListPersistentDiskResponse;
+import org.pmiops.workbench.model.Disk;
+import org.pmiops.workbench.model.EmptyResponse;
+import org.pmiops.workbench.model.WorkspaceAccessLevel;
+import org.pmiops.workbench.notebooks.LeonardoNotebooksClient;
+import org.pmiops.workbench.utils.mappers.LeonardoMapper;
+import org.pmiops.workbench.workspaces.WorkspaceAuthService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class DiskController implements DiskApiDelegate {
+  private static final Logger log = Logger.getLogger(DiskController.class.getName());
+  private final LeonardoNotebooksClient leonardoNotebooksClient;
+  private final WorkspaceDao workspaceDao;
+  private final WorkspaceAuthService workspaceAuthService;
+  private final LeonardoMapper leonardoMapper;
+  private final Provider<DbUser> userProvider;
+
+  @Autowired
+  public DiskController(
+      LeonardoNotebooksClient leonardoNotebooksClient,
+      WorkspaceAuthService workspaceAuthService,
+      WorkspaceDao workspaceDao,
+      Provider<DbUser> userProvider,
+      LeonardoMapper leonardoMapper) {
+    this.leonardoNotebooksClient = leonardoNotebooksClient;
+    this.workspaceAuthService = workspaceAuthService;
+    this.workspaceDao = workspaceDao;
+    this.userProvider = userProvider;
+    this.leonardoMapper = leonardoMapper;
+  }
+
+  @Override
+  public ResponseEntity<Disk> getDisk(String workspaceNamespace) {
+    DbWorkspace dbWorkspace = lookupWorkspace(workspaceNamespace);
+    String firecloudWorkspaceName = dbWorkspace.getFirecloudName();
+    String googleProject = dbWorkspace.getGoogleProject();
+    workspaceAuthService.enforceWorkspaceAccessLevel(
+        workspaceNamespace, firecloudWorkspaceName, WorkspaceAccessLevel.WRITER);
+    workspaceAuthService.validateActiveBilling(workspaceNamespace, firecloudWorkspaceName);
+    try {
+      LeonardoGetPersistentDiskResponse response;
+      String pdNamePrefix = userProvider.get().getUserPDNamePrefix();
+      List<LeonardoListPersistentDiskResponse> responseList =
+          leonardoNotebooksClient.listPersistentDiskByProject(googleProject, false)
```

Comment:
I would also really like to log here if we see a case where there are multiple active PDs for a single user... this indicates we've leaked a PD